dev_docs/tutorials/data/search.mdx

```diff
@@ -0,0 +1,490 @@
+
+First of all, in your plugin's `start` lifecycle method, call the `enableStorage` method. This method helps the `Session Service` gather all information it requires to `save` the `Search Sessions` upon a user's request, as well as constructing the `restore` state:
+
+```ts
+export class MyPlugin implements Plugin {
+  public start(core: CoreStart, { data }: MyPluginStartDependencies) {
+    const sessionRestorationDataProvider: SearchSessionInfoProvider = {
+      data,
+      getDashboard
+    }
+
+    data.search.session.enableStorage({
+      getName: async () => {
+        // return the name you want to give the saved Search Session
+        return `MyApp_${Math.random()}`;
+      },
+      getUrlGeneratorData: async () => {
+        // Read more about the URL generators **link**
+        return {
+          urlGeneratorId: MY_URL_GENERATOR,
+          initialState: getUrlGeneratorState({ ...deps, shouldRestoreSearchSession: false }),
+          restoreState: getUrlGeneratorState({ ...deps, shouldRestoreSearchSession: true }),
+        };
+      },
+    });
+  }
+}
+```
+
+<DocCallOut title="Search Sessions Restore State">
+  The restore state of a Search Session may be different from the initial state used to create it. For example, where the initial state may contain relative dates, in the restore state those must be converted to absolute ones (Read more about the [NowProvider here]()).  
+</DocCallOut>
+
+<DocCallOut color="danger" title="Search Sessions Indicator">
+  Calling `enableStorage` will also enable the `Search Session Indicator` component in the chrome component of your solution. The `Search Session Indicator` is a small button, used by default to engage users and save new Search Sessions. If you wish to implement your own UI, contact us to decouple this behavior.  
+</DocCallOut>
+
+##### Start a new Search Session 
+
+Make sure to call `start` when the *state* you previously defined changes. 
+
+```ts
+
+function onSearchSessionConfigChange() {
+  this.searchSessionId = data.search.sessions.start();
+}
+
+```
+
+Pass the `searchSessionId` to every `search` call inside your application. If you're using `Embeddables` make sure to pass down the `searchSessionId` as `input`. 
+
+If for any reason, you can't pass the `searchSessionId` directly, you can also retrieve it from the service.
+
+```ts
+const currentSearchSessionId = data.search.sessions.getSessionId();
+
+```
+
+##### Clear Search Sessions 
+
+Creating a new `Search Session` clears the previous one. However, be sure to explicitly `clear` the `Search Session` when your application is being destroyed:
+
+```ts
+function onDestroy() {
+  data.search.session.clear();
+}
+
+```
+
+If you don't call `clear`, you will see a warning in the console while developing. However, when running in production, you will get a fatal error. This is done to avoid leakage of unrelated `search` requests into an existing `Search Session`, left open by mistake.
```

Comment:
```suggestion
If you don't call `clear`, you will see a warning in the console while developing. However, when running in production, you will get a fatal error. This is done to avoid leakage of unrelated search requests into an existing search session left open by mistake.
```